# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/locale/LocaleService.cpp
# Commit: 7ae4df10749c
# Full Hash: 7ae4df10749c176009923591c252c634571c21d1
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-02 15:21:09
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r ece9a4223349 -r 7ae4df10749c intl/locale/LocaleService.cpp
--- a/intl/locale/LocaleService.cpp	Thu Jul 02 03:39:46 2020 +0000
+++ b/intl/locale/LocaleService.cpp	Thu Jul 02 02:51:41 2020 +0000
@@ -295,13 +295,14 @@
 
 static bool GetGREFileContents(const char* aFilePath, nsCString* aOutString) {
   // Look for the requested file in omnijar.
-  RefPtr<nsZipArchive> zip = Omnijar::GetReader(Omnijar::GRE);
+  RefPtr<CacheAwareZipReader> zip = Omnijar::GetReader(Omnijar::GRE);
   if (zip) {
-    nsZipItemPtr<char> item(zip, aFilePath);
-    if (!item) {
+    uint32_t length;
+    const uint8_t* data = zip->GetData(aFilePath, &length);
+    if (!data) {
       return false;
     }
-    aOutString->Assign(item.Buffer(), item.Length());
+    aOutString->Assign(reinterpret_cast<const char*>(data), length);
     return true;
   }
 